internal/core/software_renderer.rs

```diff
@@ -545,22 +676,24 @@ fn render_window_frame_by_line(
     renderer: &SoftwareRenderer,
     mut line_buffer: impl LineBufferProvider,
 ) -> PhysicalRegion {
+    let rotation = RotationInfo { orientation: renderer.rotation.get(), screen_size: size };
     let mut scene = prepare_scene(window, size, renderer);
 
     let dirty_region = scene.dirty_region;
+    let transformed = dirty_region.transformed(rotation);
```

Comment:
> Even better of course would be to overwrite dirty_region, if that's possible.

That's not possible since we need to return it.  Now the question is whether we should return it as transformed or not.   And I do think it actually makes sense to return it as transformed if we say that the Physical coordinate are indeed transformed.